File: 인범/2026_04_29/word_chain.py
def solution(n, words):
    from collections import defaultdict
    mydict = defaultdict(int)
    answer = []
    mywords = set()
    curr = 1
    passing = True
    
    for idx, word in enumerate(words):
        if idx>0 and word[0] != words[idx-1][-1]:
            # print(word[0], words[idx-1][-1])
            mydict[curr] += 1
            passing = False
            break
        if word not in mywords:
            # print(word)
            mywords.add(word)
            mydict[curr] += 1
            if curr == n : curr = 1
            else : curr += 1
        else:
            # print(word)
            mydict[curr] += 1
            passing = False
            break;
    
    if not passing :
        answer.append(curr)
        answer.append(mydict[curr])
        return answer
    else : return [0,0]

File: 인범/2026_04_29/word_chain.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int curr=1, cyc=1;
        int index = 0;
        boolean flag = true;
        Set<String> set = new HashSet<>();
        
        while(flag){
            String word = words[index];
            if(index==0){
                set.add(word);
                curr++;
                index++;
            }
            else{ // 첫 단어 이후
                String before = words[index-1];
                if(before.charAt(before.length()-1) != word.charAt(0)){
                    flag = false;
                }
                else if(!set.add(word)){ // 이미 잇는 경우 boolean false return
                    flag = false;
                }
                else{ // set 에 자료 추가 완료
                    curr++;
                    index++;
                }
            }
            if(curr>n){
                curr=1;
                cyc++;
            }
            if(index>=words.length){
                flag=false;
            }
        }
        if(!(index>=words.length)){
        int[] answer = {curr,cyc};
        return answer;
        }
        else{
            int[] answer = {0,0};
            return answer;
        }
    }
}